docs/high-availability.md

```diff
@@ -0,0 +1,199 @@
+
+
+## Replicated API Servers
+Once you have replicated etcd set up correctly, we will also install the apiserver using the kubelet.
+
+### Installing configuration files
+First you need to create the initial log file, so that Docker mounts a file instead of a directory:
+
+```
+touch /var/log/kube-apiserver.log
+```
+
+Next, you need to create a ```/srv/kubernetes/``` directory on each node.  This directory includes:
+   * basic_auth.csv  - basic auth user and password
+   * ca.crt - Certificate Authority cert
+   * known_tokens.csv - tokens that entities (e.g. the kubelet) can use to talk to the apiserver
+   * kubecfg.crt - Client certificate, public key
+   * kubecfg.key - Client certificate, private key
+   * server.cert - Server certificate, public key
+   * server.key - Server certificate, private key
+
+The easiest way to create this directory, may be to copy it from the master node of a working cluster, or you can manually generate these files yourself.
+
+### Starting the API Server
+Once these files exist, copy the [kube-apiserver.manifest](high-availability/kube-apiserver.manifest) into ```/etc/kubernetes/manifests/``` on each master node.
+
+The kubelet monitors this directory, and will automatically create an instance of the ```kube-apiserver``` container using the pod definition specified
+in the file.
+
+### Load balancing.
+At this point, you should have 3 apiservers all working correctly.  If you set up a network load balancer, you should
+be able to access your cluster via that load balancer, and see traffic balancing between the apiserver instances.  Setting
+up a load balancer will depend on the specifics of your platform, for example instructions for the Google Cloud
+Platform can be found [here](https://cloud.google.com/compute/docs/load-balancing/)
+
+Note, if you are using authentication, you may need to regenerate your certificat to include the IP address of the balancer,
+in addition to the IP addresses of the individual nodes.
+
+For pods that you deploy into the cluster, the ```kubernetes``` service/dns name should provide a load balanced endpoint for the master automatically.
+
+For external users of the API (e.g. the ```kubectl``` command line interface, continuous build pipelines, or other clients) you will want to configure
+them to talk to the external load balancer's IP address.
+
+## Master-elected components
+So far we have set up state storage, and we have set up the API server, but we haven't run anything that actually modifies
+cluster state, such as the controller manager and scheduler.  To achieve this reliably, we only want to have one actor modifying state at a time, but we want replicated
+instances of these actors, in case a machine dies.  To achieve this, we are going to use a lease-lock in etcd to perform
+master election.  We'll use a simple utility that runs the master election, and then either creates or deletes a pod
```

Comment:
total aside - i always thought the word "election" was kinda bogus in this context, it's closer to sword in the stone type selection than an election